src/Microsoft.Health.Fhir.Core/Features/Persistence/ResourceWrapperFactory.cs

```diff
@@ -77,5 +87,68 @@ public ResourceWrapper Create(ResourceElement resource, bool deleted, bool keepM
                 _claimsExtractor.Extract(),
                 searchParamHash);
         }
+
+        /// <inheritdoc />
+        public void Update(ResourceWrapper resourceWrapper)
+        {
+            var resourceElement = _resourceDeserializer.Deserialize(resourceWrapper);
+            var newIndices = _searchIndexer.Extract(resourceElement);
+            ExtractMinAndMaxValues(newIndices);
+            resourceWrapper.UpdateSearchIndices(newIndices);
+        }
+
+        // A given search parameter can have multiple values. We want to keep track of which
+        // of these values are the min and max for each parameter and mark the corresponding
+        // SearchValue object appropriately.
+        private void ExtractMinAndMaxValues(IReadOnlyCollection<SearchIndexEntry> searchIndices)
+        {
+            var minValues = new Dictionary<Uri, ISupportSortSearchValue>();
+            var maxValues = new Dictionary<Uri, ISupportSortSearchValue>();
+
+            foreach (SearchIndexEntry currentEntry in searchIndices)
+            {
+                // Currently we are tracking the min/max values only for string type parameters.
+                var currentValue = currentEntry.Value as ISupportSortSearchValue;
+                if (currentValue == null)
+                {
+                    continue;
+                }
+
+                if (minValues.TryGetValue(currentEntry.SearchParameter.Url, out ISupportSortSearchValue existingMinValue))
+                {
+                    if (currentValue.CompareTo(existingMinValue, ComparisonRange.Min) < 0)
+                    {
+                        minValues[currentEntry.SearchParameter.Url] = currentValue;
+                    }
+                }
+                else
+                {
+                    minValues.Add(currentEntry.SearchParameter.Url, currentValue);
+                }
+
+                if (maxValues.TryGetValue(currentEntry.SearchParameter.Url, out ISupportSortSearchValue existingMaxValue))
+                {
+                    if (currentValue.CompareTo(existingMaxValue, ComparisonRange.Max) > 0)
+                    {
+                        maxValues[currentEntry.SearchParameter.Url] = currentValue;
+                    }
+                }
+                else
+                {
+                    maxValues.Add(currentEntry.SearchParameter.Url, currentValue);
+                }
+            }
+
+            // Update the actual StringSearchValue objects with the appropriate IsMin/IsMax value
```

Comment:
>StringSearchValue [](start = 33, length = 17)

nit: comment needs to be updated to ISupportSearchValue (or maybe we can remove the comment itself).